core/src/main/java/tech/pegasys/web3signer/core/service/jsonrpc/handlers/signing/TransactionSerializer.java

```diff
@@ -73,11 +76,12 @@ private static byte[] prependEip1559TransactionType(byte[] bytesToSign) {
         .array();
   }
 
-  private SignatureData sign(final byte[] bytesToSign) {
-    final SecpArtifactSignature artifactSignature =
-        (SecpArtifactSignature) signer.sign(Bytes.of(bytesToSign));
+  private SignatureData sign(final String eth1Address, final byte[] bytesToSign) {
+    final String bytesSigned =
+        secpSigner.sign(normaliseIdentifier(eth1Address), Bytes.of(bytesToSign)).orElseThrow();
```

Comment:
Adding a sign method that returns an ArtifactSignature would be a nicer way of doing this rather than converting to a hex string and back to Bytes. We already have the SignerIdentifier using generics so it could return a SecpArtifactSignature we could use directly.

Reply:
done